app/jobs/batch_operation_process_one_job.rb

```diff
@@ -17,4 +23,10 @@ def perform(batch_operation, dossier)
   rescue ActiveRecord::RecordNotFound
     dossier.update_column(:batch_operation_id, nil)
   end
+
+  private
+
+  def called_for_last_time?(job)
+    job.arguments.first.dossiers.count.zero?
```

Comment:
j'ai l'impression qu'on deplace le problème de concurrence un peu plus loin dans la chaine. ce petit reload me rassurerait 
```suggestion
    job.arguments.first.reload.dossiers.count.zero?
```